src/main/scala/pipedsl/Parser.scala

```diff
@@ -170,15 +212,16 @@ class Parser extends RegexParsers with PackratParsers {
   lazy val binAnd: P[Expr] = parseOp(binBOr, and)
   lazy val binOr: P[Expr] = parseOp(binAnd, or)
   lazy val binConcat: P[Expr] = parseOp(binOr, concat)
-  lazy val expr: Parser[Expr] = positioned(binConcat)
+  lazy val nontern: Parser[Expr] = positioned(binConcat)
+  lazy val expr :Parser[Expr] = ternary | nontern
+
+     
   lazy val lhs: Parser[Expr] = memAccess | variable
 
   lazy val simpleCmd: P[Command] = positioned {
     speccall |
-    typ.? ~ variable ~ "=" ~ expr ^^ { case t ~ n ~ _ ~ r => n.typ = t; CAssign(n, r) } |
-      typ.? ~ lhs ~ "<-" ~ expr ^^ { case t ~ l ~ _ ~ r => l.typ = t
-        CRecv(l, r)
-      } |
+    typ.? ~ variable ~ "=" ~ expr ^^ { case t ~ n ~ _ ~ r =>  n.typ = t; CAssign(n, r, t) } |
```

Comment:
Is there a reason you  generate TypeVars for bitwidths but not for assigned types?
Like `CAssign` could take a `Type` instead of a `Maybe` as its third parameter and fill it with a type variable

Reply:
removed extra param for CRecv/CAssign